src/build/cargo.rs

```diff
@@ -248,7 +251,12 @@ fn run_cargo(
             );
         }
         Err(e) => {
-            debug!("Error running compile_with_exec: {:?}", e);
+            if !reached_primary.load(Ordering::SeqCst) {
+                debug!("Error running compile_with_exec: {:?}", e);
+                return Err(e);
+            } else {
+                warn!("Ignoring error running compile_with_exec: {:?}", e);
```

Comment:
I found this a bit strange we were ignoring cargo errors. I initially had it always propagating the error, which seemed to work fine and make sense.

However `cmd_test_complete_self_crate_name` started to fail getting:
```
"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[{"message":"failed to stat `/home/alex/project/rls/target/rlsit/t2/ws_with_test_dir/target/rls/debug/.fingerprint/library-41efb745a4eb321b/dep-test-integration-test-test-41efb745a4eb321b`\nNo such file or directory (os error 2)","range":{"end":{"character":0,"line":9999},"start":{"character":0,"line":0}},"severity":1}],"uri":"file:///home/alex/project/rls/target/rlsit/t2/ws_with_test_dir/Cargo.toml"}}Content-Length: 104
```

Why this error happens I'm not sure, I added the extra `if !reached_primary.load(Ordering::SeqCst) {` to act as we did previously ie ignoring this error as before.

I'm not sure why it's ok to ignore errors if we've "reached_primary".

Reply:
I think because if we're building the primary crate, then we'll report errors in the usual way, rather than doing the Cargo-specific reporting.